src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -74,8 +107,190 @@ public static void generate(String path, OutputStream out, String... moduleFolde
+            if (detailNode.getType() != JavadocTokenTypes.LEADING_ASTERISK
+                    && !TOKEN_TEXT_PATTERN.matcher(childText).matches()) {
+                result.append(childText);
+            }
+        }
+        return result.toString().trim();
+    }
+
+    /**
+     * Extracts description nodes from javadoc.
+     *
+     * @param javadoc the Javadoc to extract the description from.
+     * @return the description nodes of the setter.
+     */
+    private static List<DetailNode> getDescriptionNodes(DetailNode javadoc) {
+        final DetailNode[] children = javadoc.getChildren();
+        final List<DetailNode> descriptionNodes = new ArrayList<>();
+        for (final DetailNode child : children) {
+            if (isEndOfDescription(child)) {
+                break;
+            }
+            descriptionNodes.add(child);
+        }
+        return descriptionNodes;
+    }
+
+    /**
+     * Determines if the given child index is the end of the description. The end of the
+     * description is defined as 4 consecutive nodes of type NEWLINE, LEADING_ASTERISK, NEWLINE,
+     * LEADING_ASTERISK. This is an asterisk that is alone on a line. Just like the one below
+     * this line.
+     *
+     * @param child the child to check.
+     * @return true if the given child index is the end of the description.
+     */
+    private static boolean isEndOfDescription(DetailNode child) {
+        final DetailNode nextSibling = JavadocUtil.getNextSibling(child);
+        final DetailNode secondNextSibling = JavadocUtil.getNextSibling(nextSibling);
+        final DetailNode thirdNextSibling = JavadocUtil.getNextSibling(secondNextSibling);
+
+        return child.getType() == JavadocTokenTypes.NEWLINE
+                    && nextSibling.getType() == JavadocTokenTypes.LEADING_ASTERISK
+                    && secondNextSibling.getType() == JavadocTokenTypes.NEWLINE
+                    && thirdNextSibling.getType() == JavadocTokenTypes.LEADING_ASTERISK;
+    }
+
+    /**
+     * Get the module name from the file. The module name is the file name without the extension.
+     * If the module name ends with "Check", then it's removed.
+     *
+     * @param file file to extract the module name from.
+     * @return module name.
+     */
+    private static String getModuleName(File file) {
+        final String fullFileName = file.getName();
+        return CommonUtil.getFileNameWithoutExtension(fullFileName);
+    }
+
+    /**
+     * Get module type(check/filter/filefilter) based on module name.
+     *
+     * @param moduleName module name.
+     * @return module type.
+     */
+    private static ModuleType getModuleType(String moduleName) {
+        final ModuleType result;
+        if (moduleName.endsWith("FileFilter")) {
+            result = ModuleType.FILEFILTER;
+        }
+        else if (moduleName.endsWith("Filter")) {
+            result = ModuleType.FILTER;
+        }
+        else {
+            result = ModuleType.CHECK;
```

Comment:
Are you sure in this ? 
All Checks has name suffix "Check". 
There are more modules like Holder, that we s bad hack but still we have them.

Reply:
Holders are still checks.

Reply:
Ok, we will deal with this later on